kolibri/plugins/learn/assets/src/views/TopicsPage.vue

```diff
@@ -527,12 +526,23 @@
         return this.channel.name;
       },
       resources() {
-        const resources = this.contents.filter(content => content.kind !== ContentNodeKinds.TOPIC);
-        // If there are no topics, then just display all resources we have loaded.
-        if (!this.topics.length) {
-          return resources;
+        return this.contents.filter(content => content.kind !== ContentNodeKinds.TOPIC);
+      },
+      resourcesDisplayed() {
+        // if no folders are shown at this level, show more resources to fill the space
+        if (this.topics.length == 0 && !this.showMoreResources) {
+          return this.resources.slice(0, this.childrenToDisplay * 2);
+          // if there are topics, and the user has not clicked show more
+          // default to just the preview number
+        } else if (this.topics.length > 0 && !this.showMoreResources) {
+          return this.resources.slice(0, this.childrenToDisplay);
+          // otherwise display all resources
+        } else {
+          return this.resources;
```

Comment:
Stylistic nit - I prefer to return at the end of the function, rather than in a redundant `else` statement.